core/src/main/java/io/grpc/transport/ServerListener.java

```diff
@@ -44,5 +43,12 @@
    * @param transport the new transport to be observed.
    * @return a listener for stream creation events on the transport.
    */
-  ServerTransportListener transportCreated(Service transport);
+  ServerTransportListener transportCreated(ServerTransport transport);
+
+  /**
+   * The server is shutting down. No new transports will be processed, but existing streams may
+   * continue. Shutdown is only caused by a call to {@link Server#shutdown()}. All resources have
+   * been released.
+   */
+  void serverShutdown();
```

Comment:
Should this be called `serverShuttingDown` to match the comments? `serverShutdown` implies that it's been fully shutdown, which I don't think is the case.